cmd/loadOutputTranslation.go

```diff
@@ -0,0 +1,236 @@
+// Copyright © Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/Azure/azure-storage-azcopy/ste"
+
+	"github.com/Azure/azure-storage-azcopy/common"
+)
+
+const (
+	clfsErrTag            = "ERR"
+	clfsInitializationTag = "AZCOPY-CONFIG:"
+	clfsInfoTag           = "AZCOPY-INFO:"
+	clfsProgressTag       = "AZCOPY:"
+	clfsExitTag           = "AZCOPY-FINAL:"
+)
+
+type clfsExtensionOutputParser struct {
+	lcm          common.LifecycleMgr
+	finishSignal chan interface{} // TODO consider using a wait group
+}
+
+func newClfsExtensionOutputParser(lcm common.LifecycleMgr) *clfsExtensionOutputParser {
+	return &clfsExtensionOutputParser{lcm: lcm, finishSignal: make(chan interface{}, 1)}
+}
+
+// should be started in another go-routine
+func (c *clfsExtensionOutputParser) startParsing(reader io.Reader) {
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if strings.Contains(line, clfsExitTag) {
+			c.processEndOfJob(line)
+		} else if strings.Contains(line, clfsInitializationTag) {
+			c.processInitialization(line)
+		} else if strings.Contains(line, clfsInfoTag) {
+			c.processInfo(line)
+		} else if strings.Contains(line, clfsProgressTag) {
+			c.processProgress(line)
+		} else if strings.Contains(line, clfsErrTag) {
+			c.processError(line)
```

Comment:
Consider doing strings.HasPrefix instead of strings.Contains. It's unlikely that we'll see any of these other strings inside of an output line, but it would prevent us from re-running something like processEndOfJob.